ztp/tools/pgt2acm/main.go

```diff
@@ -0,0 +1,357 @@
+	}
+	if sourceCRListString != nil {
+		preRenderSourceCRList = strings.Split(*sourceCRListString, ",")
+	}
+	return preRenderPatchKindList, preRenderSourceCRList
+}
+
+func main() {
+	// Defines the input PGT directory or file
+	var inputFile = flag.String("i", "", "the PGT input file")
+	// Defines the output directory for generated ACM templates
+	var outputDir = flag.String("o", "", "the ACMGen output Directory")
+	// Defines the input schema file. Schema allows patching CRDs containing lists of objects
+	var schema = flag.String("s", "", "the optional schema for all non base CRDs")
+	// Defines list of manifest kinds to which to pre-render patches to
+	var preRenderPatchKindString = flag.String("k", "", "the optional list of manifest kinds for which to pre-render patches")
+	// Optionally generates ACM policies for PGT and ACM Gen templates
+	var generateACMPolicies = flag.Bool("g", false, "optionally generates ACM policies for PGT and ACM Gen templates")
+	// Defines ns.yaml file for templates
+	var NSYAML = flag.String("n", fileutils.NamespaceFileName, "the optional ns.yaml file path")
+	// optionally disables generating default placement in ns.yaml
+	var skipDefaultPlacementBindings = flag.Bool("p", false, "optionally disable generating default placement bindings in ns.yaml")
+	// Defines source-crs directory location
+	var sourceCRs = flag.String("c", "", "the optional comma delimited list of reference source CRs templates")
+	// Optionally generate placement API template containing toleration for
+	//     - effect: NoSelect
+	//       key: cluster.open-cluster-management.io/unreachable
+	var workaroundPlacement = flag.Bool("w", false, "Optional workaround to generate placement API template containing cluster.open-cluster-management.io/unreachable toleration")
+
+	preRenderPatchKindList, preRenderSourceCRList := processFlags(inputFile, outputDir, preRenderPatchKindString, sourceCRs)
+
+	var err error
+	if sourceCRs != nil && *sourceCRs != "" {
+		err = fileutils.CopySourceCrs(*inputFile, *outputDir, preRenderSourceCRList)
+		if err != nil {
+			fmt.Printf("Could not copy source-crs files, err: %s", err)
+			os.Exit(1)
+		}
+	}
+
+	allFilesInInputPath, err := fileutils.GetAllYAMLFilesInPath(*inputFile)
+	if err != nil {
+		fmt.Printf("Could not get file list, err: %s", err)
+		os.Exit(1)
+	}
+	// convert all PGT files
+	err = convertAllPGTFiles(preRenderPatchKindList, allFilesInInputPath, inputFile, outputDir, schema, workaroundPlacement)
+	if err != nil {
+		fmt.Printf("Could not convert PGT files, err: %s", err)
+		os.Exit(1)
+	}
+
+	if NSYAML != nil && *NSYAML != "" && skipDefaultPlacementBindings != nil {
+		err = fileutils.CopyAndProcessNSAndKustomizationYAML(*NSYAML, *inputFile, *outputDir, *skipDefaultPlacementBindings)
+		if err != nil {
+			fmt.Printf("Could not post-process %s and %s files, err: %s", *NSYAML, fileutils.KustomizationFileName, err)
+		}
+	}
+
+	if generateACMPolicies != nil && *generateACMPolicies {
```

Comment:
For flags of `Bool` type I think it's enough to check `if *flag` since they all have defaults to `false`.

Reply:
Fixed